### DISCUSSION TITLE
Error when trying to decrypt

### DISCUSSION THREAD

**Discussion by SymphonySimper** [twpayne/chezmoi#3147] [Q&A] [answered]
This what I get when I try to run initialize my [dotfiles](https://github.com/SymphonySimper/dotfiles) with chezmoi.
I ran this command with bash shell which comes with `git` aka `git bash`. 
- [X] I have read [chezmoi's user guide](https://chezmoi.io/user-guide/command-overview/), and not found the answer.
- [X] I have searched [chezmoi's reference guide](https://chezmoi.io/reference/), and not found the answer.

Output of: ` chezmoi --verbose init --apply SymphonySimper`
```sh
diff --git a/decrypt-private-key.sh b/decrypt-private-key.sh
new file mode 100755
index 0000000000000000000000000000000000000000..cfaa57eb529481d092a72bdc481cc3bb5135da67
--- /dev/null
+++ b/decrypt-private-key.sh
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+if [ ! -f "$HOME/key.txt" ]; then
+    age --decrypt --output "$HOME/key.txt" "C:/Users/symph/.local/share/chezmoi/key.txt.age"
+    chmod 600 "$HOME/key.txt"
+fi
\ No newline at end of file
chezmoi: fork/exec C:\Users\symph\AppData\Local\Temp\2282338352.decrypt-private-key.sh: %1 is not a valid Win32 application.
```
Output of: `chezmoi doctor`
```sh
RESULT   CHECK                MESSAGE
ok       version              v2.36.1, commit 0b0fa89f612f5f885ccfc8c3890f515cc668c5c6, built at 2023-07-30T12:33:19Z, built by goreleaser
ok       latest-version       v2.36.1
ok       os-arch              windows/amd64
ok       systeminfo           Microsoft Windows 11 Home Single Language (10.0.22621 N/A Build 22621)
ok       go-version           go1.20.6 (gc)
ok       executable           ~/scoop/apps/chezmoi/current/chezmoi.exe
ok       upgrade-method       replace-executable
ok       config-file          ~/.config/chezmoi/chezmoi.toml, last modified 2023-08-03T18:53:55+05:30
ok       source-dir           ~/.local/share/chezmoi is a git working tree (clean)
ok       suspicious-entries   no suspicious entries
ok       working-tree         ~/.local/share/chezmoi is a git working tree (clean)
ok       dest-dir             ~ is a directory
ok       cd-command           found ~/scoop/apps/git/2.41.0.3/usr/bin/bash.exe
ok       cd-args              'C:\\Users\\symph\\scoop\\apps\\git\\2.41.0.3\\usr\\bin\\bash.exe'
info     diff-command         not set
ok       edit-command         found C:/Windows/system32/notepad.exe
ok       edit-args            'C:\\Windows\\system32\\notepad.exe'
ok       git-command          found ~/scoop/apps/git/2.41.0.3/mingw64/bin/git.exe, version 2.41.0
ok       merge-command        found ~/scoop/apps/git/2.41.0.3/usr/bin/vimdiff.exe
ok       shell-command        found ~/scoop/apps/git/2.41.0.3/usr/bin/bash.exe
ok       shell-args           'C:\\Users\\symph\\scoop\\apps\\git\\2.41.0.3\\usr\\bin\\bash.exe'
ok       age-command          found ~/scoop/shims/age.exe, version 1.1.1
ok       gpg-command          found ~/scoop/apps/git/2.41.0.3/usr/bin/gpg.exe, version 2.2.41
info     pinentry-command     not set
info     1password-command    op not found in $PATH
info     bitwarden-command    bw not found in $PATH
info     dashlane-command     dcli not found in $PATH
info     gopass-command       gopass not found in $PATH
info     keepassxc-command    keepassxc-cli not found in $PATH
info     keepassxc-db         not set
info     keeper-command       keeper not found in $PATH
info     lastpass-command     lpass not found in $PATH
info     pass-command         pass not found in $PATH
info     passhole-command     ph not found in $PATH
info     rbw-command          rbw not found in $PATH
info     vault-command        vault not found in $PATH
info     vlt-command          vlt not found in $PATH
info     secret-command       not set
```

**Comment by twpayne** [comment]
You need to include much more information if you want people to help you.

See the [support request issue template](https://github.com/twpayne/chezmoi/blob/master/.github/ISSUE_TEMPLATE/01_support_request.md) and do some research on how to write a good support request or bug report.

**Reply by SymphonySimper** [reply]
My bad I will add the relevant details.

**Reply by SymphonySimper** [reply]
I have updated my discussion question. I hope the info will be helpful. Thanks.

**Comment by twpayne** [comment, marked as answer]
Your `decrypt-private-key.sh` script is a shell script that only runs on UNIX-like systems. It doesn't work on Windows.

On Windows, you need to disable the script and either install your private key manually or write an equivalent Windows batch script or PowerShell script.

**Reply by SymphonySimper** [reply]
Yep, I understand that. But is there any way that I could tell `chezmoi` to use bash to run the script. As you can you bash shell in windows provided by *git bash* to run most of the shell scripts without major changes.

**Reply by twpayne** [reply]
It might be possible to do this using chezmoi's [interpreters feature](https://www.chezmoi.io/reference/target-types/#scripts-on-windows).

**Reply by SymphonySimper** [reply]
Thanks. But that did not work as expected. So I just wrote powershell script.

**Reply by twpayne** [reply]
Did this not work as expected due to something in chezmoi or something in the script?

**Reply by SymphonySimper** [reply]
I'm not 100% sure. Bash complains that file(C:\Users\symph\AppData\Local\Temp\2282338352.decrypt-private-key.sh) doesn't exist.

**Reply by twpayne** [reply]
Can `git bash` execute other shell scripts in `C:\Users\symph`? Does it need an absolute or relative path? Can it handle backslashes in filenames or do they need escaping or converting to forward slashes?

**Reply by SymphonySimper** [reply]
Sorry for the delayed response.
Seems like the path is the issue. `git bash` requires UNIX like path. So instead of `C:\Users\symph` it requires the path to be as `/mnt/c/Users/symph`. But if you run the script by launch the shell (`git bash`). Even *Windows* style paths with `\\` ,`\`,`/` are working properly.

**Reply by twpayne** [reply]
Thanks for the investigation. It seems like `git bash` cannot be used as a substitute for `bash` on Windows, so the best option is to use a PowerShell or batch script on Windows.

**Reply by SymphonySimper** [reply]
Sure no problem. Thanks for you time and making an awesome dotfiles manager.
